java/vector/src/main/java/org/apache/arrow/vector/compare/ApproxEqualsVisitor.java

```diff
@@ -102,9 +107,9 @@ protected boolean compareUnionVectors(UnionVector left) {
       return false;
     }
 
+    ApproxEqualsVisitor visitor = new ApproxEqualsVisitor(floatEpsilon, doubleEpsilon);
     for (int k = 0; k < leftChildren.size(); k++) {
-      ApproxEqualsVisitor visitor = new ApproxEqualsVisitor(rightChildren.get(k),
-          floatEpsilon, doubleEpsilon);
+      visitor.set(rightChildren.get(k), 0, 0, rightChildren.get(k).getValueCount(), true);
       if (!leftChildren.get(k).accept(visitor, null)) {
```

Comment:
Should we give a message if visitor is directly used without setting params?

Reply:
Sounds good. Thanks.

Reply:
I have updated the code to make leftStart, rightStart and length have -1 as default value. Since each visit method calls validate method first, if the set method is not called, an exception will be thrown.